src/test/java/com/fasterxml/jackson/databind/jsontype/TestSubtypes.java

```diff
@@ -216,8 +239,12 @@ public void testSerialization() throws Exception
         mapper.registerSubtypes(new NamedType(SubB.class, "typeB"));
         assertEquals("{\"@type\":\"typeB\",\"b\":1}", mapper.writeValueAsString(bean));
 
+        // the first registered type name is used for serialization
```

Comment:
It'd probably be better to leave existing test as is, but also add a new extended one.